includes/multi-currency/Analytics.php

```diff
@@ -0,0 +1,232 @@
+		],
+	];
+
+	/**
+	 * Instance of MultiCurrency.
+	 *
+	 * @var MultiCurrency $multi_currency
+	 */
+	private $multi_currency;
+
+	/**
+	 * Constructor
+	 *
+	 * @param MultiCurrency $multi_currency Instance of MultiCurrency.
+	 */
+	public function __construct( MultiCurrency $multi_currency ) {
+		$this->multi_currency = $multi_currency;
+		$this->init();
+	}
+
+	/**
+	 * Initialise all actions, filters and hooks related to analytics support.
+	 *
+	 * @return void
+	 */
+	public function init() {
+		if ( is_admin() ) {
+			add_filter( 'admin_enqueue_scripts', [ $this, 'enqueue_admin_scripts' ] );
+		}
+
+		// TODO: Remove this or make it only apply in dev mode.
+		add_filter( 'woocommerce_analytics_report_should_use_cache', [ $this, 'disable_report_caching' ] );
+
+		// If we aren't making a REST request, return before adding these filters.
+		if ( ! WC()->is_rest_api_request() ) {
+			return;
+
+		}
+
+		add_filter( 'woocommerce_analytics_clauses_select', [ $this, 'filter_select_clauses' ], self::PRIORITY_DEFAULT, 2 );
+		add_filter( 'woocommerce_analytics_clauses_join', [ $this, 'filter_join_clauses' ] );
+	}
+
+	/**
+	 * Register the CSS and JS scripts.
+	 *
+	 * @return void
+	 */
+	public function register_admin_scripts() {
+		$script_src_url    = plugins_url( 'dist/multi-currency-analytics.js', WCPAY_PLUGIN_FILE );
+		$script_asset_path = WCPAY_ABSPATH . 'dist/multi-currency-analytics.asset.php';
+		$script_asset      = file_exists( $script_asset_path ) ? require_once $script_asset_path : [ 'dependencies' => [] ];
+
+		wp_register_script(
+			self::SCRIPT_NAME,
+			$script_src_url,
+			$script_asset['dependencies'],
+			\WC_Payments::get_file_version( 'dist/multi-currency-analytics.js' ),
+			true
+		);
+
+		wp_register_style(
+			self::SCRIPT_NAME,
+			plugins_url( 'dist/multi-currency-analytics.css', WCPAY_PLUGIN_FILE ),
+			[ 'wc-components' ],
+			\WC_Payments::get_file_version( 'dist/multi-currency-analytics.css' )
+		);
+	}
+
+	/**
+	 * Enqueue scripts used on the analytics WP Admin pages.
+	 *
+	 * @return void
+	 */
+	public function enqueue_admin_scripts() {
+		$this->register_admin_scripts();
+
+		wp_enqueue_script( self::SCRIPT_NAME );
+		wp_enqueue_style( self::SCRIPT_NAME );
+	}
+
+	/**
+	 * Disables report caching. Used for development testing.
+	 *
+	 * @param array $args Filter arguments.
+	 *
+	 * @return boolean
+	 */
+	public function disable_report_caching( $args ): bool {
+		return false;
+	}
+
+	/**
+	 * Add columns to get the currency and converted amount (if required).
+	 *
+	 * @param string[] $clauses - An array containing the SELECT clauses to be applied.
+	 * @param string   $context - The context in which this SELECT clause is being called.
+	 *
+	 * @return array
+	 */
+	public function filter_select_clauses( array $clauses, $context ): array {
+		// If we are unable to identify a context, just return the clauses as is.
```

Comment:
It may be worth adding a `filter` here to allow other plugins to disable the multi currency related query string replacement happening here. The only reason I can think of that they would need to do that is if they were implementing their own multi-currency analytics, or if it was causing an error with a custom report, for example.

Let me know if you have any thoughts on this!

Reply:
I don't think your queries alter any column names, only joins another table and appends a column from it to the result set. If only someone uses a column iterator to fill their tables would be affected from that, so that's something they will need to solve IMHO. Right?

Reply:
Thanks for that. I agree, I think perhaps it's something we should still consider to allow more extensibility, but I don't think it's needed for the MVP. I will create an issue to consider it. Thanks!